### PR TITLE
[Merged by Bors] - OpenShift compatibility

### PR DESCRIPTION
**Pull request by razvan**

### CODE REVIEW

File: rust/operator-binary/src/superset_controller.rs [hunk 1]
```diff
@@ -182,6 +188,18 @@ pub enum Error {
     ApplyStatus {
         source: stackable_operator::error::Error,
     },
+    #[snafu(display("failed to patch service account"))]
+    ApplyServiceAccount {
+        source: stackable_operator::error::Error,
+    },
+    #[snafu(display("failed to patch role binding: {source}"))]
```
Comment (maltesander) [inline rust/operator-binary/src/superset_controller.rs:195]:
```suggestion
    #[snafu(display("failed to patch role binding"))]
```

**Review by maltesander — CHANGES_REQUESTED** [review]
Minor things. LGTM otherwise!

**Review by maltesander — APPROVED** [review]
LGTM!

File: rust/operator-binary/src/druid_connection_controller.rs [hunk 2]
```diff
@@ -72,6 +72,14 @@ pub enum Error {
         source: stackable_superset_crd::druidconnection::Error,
         druid_connection: ObjectRef<DruidConnection>,
     },
+    #[snafu(display("failed to patch service account"))]
+    ApplyServiceAccount {
+        source: stackable_operator::error::Error,
+    },
+    #[snafu(display("failed to patch role binding: {source}"))]
```
Comment (maltesander) [inline rust/operator-binary/src/druid_connection_controller.rs:79]:
```suggestion
    #[snafu(display("failed to patch role binding"))]
```

File: rust/operator-binary/src/druid_connection_controller.rs [hunk 3]
```diff
@@ -108,6 +118,21 @@ pub async fn reconcile_druid_connection(
...
+        .await
+        .with_context(|_| ApplyServiceAccountSnafu)?;
+    client
+        .apply_patch(
+            DRUID_CONNECTION_CONTROLLER_NAME,
+            &rbac_rolebinding,
+            &rbac_rolebinding,
+        )
+        .await
+        .with_context(|_| ApplyRoleBindingSnafu)?;
```
Comment (maltesander) [inline rust/operator-binary/src/druid_connection_controller.rs:134]:
```suggestion
    let (rbac_sa, rbac_rolebinding) =
        rbac::build_rbac_resources(druid_connection.as_ref(), APP_NAME);
    client
        .apply_patch(DRUID_CONNECTION_CONTROLLER_NAME, &rbac_sa, &rbac_sa)
        .await
        .context(ApplyServiceAccountSnafu)?;
    client
        .apply_patch(
            DRUID_CONNECTION_CONTROLLER_NAME,
            &rbac_rolebinding,
            &rbac_rolebinding,
        )
        .await
        .context(ApplyRoleBindingSnafu)?;
```

**Comment by razvan** [comment]
CI: https://ci.stackable.tech/view/02%20Operator%20Tests%20(custom)/job/superset-operator-it-custom/34/

**Comment by razvan** [comment]
bors merge

**Pull request closed without merging**